File: SWEA/D2/2001. 파리 퇴치/파리 퇴치.py
T = int(input())

for test_case in range(1, T+1):
    n, m = map(int, input().split())

    arr = [list(map(int, input().split())) for _ in range(n)]

    result = 0
    for i in range(n-m+1):
        for j in range(n-m+1):
            tmp = 0
            for a in range(i, i+m):
                for b in range(j, j+m):
                    tmp += arr[a][b]

            result = max(result, tmp)

    print(f"#{test_case} {result}")

File: SWEA/D2/2001. 파리 퇴치/파리 퇴치.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class Solution {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int T = sc.nextInt();

        for (int test_case = 1; test_case <= T; test_case++) {
            int n = sc.nextInt();
            int m = sc.nextInt();

            List<List<Integer>> arr = new ArrayList<>();
            for (int i = 0; i < n; i++) {
                List<Integer> row = new ArrayList<>();
                for (int j = 0; j < n; j++) {
                    row.add(sc.nextInt());
                }
                arr.add(row);
            }

            int result = 0;
            for (int i = 0; i <= n - m; i++) {
                for (int j = 0; j <= n - m; j++) {
                    int tmp = 0;
                    for (int a = i; a < i + m; a++) {
                        for (int b = j; b < j + m; b++) {
                            tmp += arr.get(a).get(b);
                        }
                    }
                    result = Math.max(result, tmp);
                }
            }

            System.out.println("#" + test_case + " " + result);
        }
    }
}